bpf/lib/nat.h

```diff
@@ -542,18 +542,16 @@ snat_v4_icmp_rewrite_egress_embedded(struct __ctx_buff *ctx,
 			return DROP_CSUM_L4;
 #endif  /* ENABLE_SCTP */
 		case IPPROTO_ICMP: {
-			__be32 from, to;
-
 			if (ctx_store_bytes(ctx, inner_l4_off +
 						offsetof(struct icmphdr, un.echo.id),
 						&state->to_sport,
 						sizeof(state->to_sport), 0) < 0)
 				return DROP_WRITE_ERROR;
-			from = tuple->sport;
-			to = state->to_sport;
-			flags = 0; /* ICMPv4 has no pseudo-header */
-			sum_l4 = csum_diff(&from, 4, &to, 4, 0);
-			csum.offset = offsetof(struct icmphdr, checksum);
+			if (l4_csum_replace(ctx, inner_l4_off + offsetof(struct icmphdr, checksum),
+					    tuple->sport,
+					    state->to_sport,
+					    sizeof(tuple->sport)) < 0)
+				return DROP_WRITE_ERROR;
```

Comment:
I'm thinking about return `DROP_CSUM_L4` rather than `DROP_WRITE_ERROR`, as we did before at https://github.com/cilium/cilium/blob/b93af2b484cc16f05c281a2be227066783322b15/bpf/lib/nat.h#L569

Reply:
Maybe we can change this returned error in your following PR.

Reply:
Yes right It's probably what I should have used.

Reply:
@jschwinger233 @sahid what's the tradeoff, any particular reason to defer that to a subsequent PR? I'm not sure I understand why we don't just address the feedback in the PR that's adding the code. :thinking:

Reply:
@joestringer That was because Sahid said he had another PR based on this, so I thought we can defer the minor changes to that PR, so as to make things happen fast. But that was before this PR hit the CI failures, since this PR has been here for weeks, I think we should address the issues at this moment. Thanks for pointing out.

Reply:
@joestringer It's my mistake, I should have fixed both of the suggestions. I wanted to rebase the code and see if by any chance the complexity issue was gone... Sorry @jschwinger233 to have put you in that situation.

All is fixed now.